### PR TITLE
[Enhancement] Remove txn lock and add tracing for run_publish_version_task

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7173

## Problem Summary(Required) ：

For each tablet's publish_txn in a txn, they all request txn lock, which makes them run serially. This PR removes txn lock and adds some tracing for `run_publish_version_task`

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by rickif — DISMISSED** [review]

File: be/src/storage/txn_manager.cpp [hunk 1]
```diff
@@ -211,40 +209,33 @@ Status TxnManager::commit_txn(KVStore* meta, TPartitionId partition_id, TTransac
...
-            }
+    if (tablet->updates() != nullptr) {
+        StarRocksMetrics::instance()->update_rowset_commit_request_total.increment(1);
+        auto st = tablet->rowset_commit(version, rowset);
+        if (!st.ok()) {
+            StarRocksMetrics::instance()->update_rowset_commit_request_failed.increment(1);
+            return st;
+        }
+    } else {
+        // TODO(ygl): rowset is already set version here, memory is changed, if save failed
```
Comment (chaoyli) [inline be/src/storage/txn_manager.cpp:220]: TODO can be removed.

File: be/src/storage/txn_manager.cpp [hunk 2]
```diff
@@ -211,40 +209,33 @@ Status TxnManager::commit_txn(KVStore* meta, TPartitionId partition_id, TTransac
...
-            // add visible rowset to tablet
-            st = tablet->add_inc_rowset(rowset);
-            if (!st.ok() && !st.is_already_exist()) {
-                // TODO: rollback saved rowset if error?
-                LOG(WARNING) << "fail to add visible rowset to tablet. rowset_id=" << rowset->rowset_id()
-                             << ", tablet_id=" << tablet->tablet_id() << ", txn_id=" << transaction_id
-                             << ", res=" << st;
-                return st;
-            }
+    if (tablet->updates() != nullptr) {
```
[the remark this answers is not in the document]
Comment (decster) [reply be/src/storage/txn_manager.cpp:212]: they will be put into a queue, and run serially.

**Review by chaoyli — DISMISSED** [review]

**Comment by decster** [comment]
starrocks_fe_unittest

**Comment by decster** [comment]
starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by chaoyli** [comment]
run starrocks_fe_unittest

**Review by chaoyli — APPROVED** [review]

**Comment by decster** [comment]
run starrocks_be_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Review by sduzh — APPROVED** [review]

**Comment by decster** [comment]
@mergify backport branch-2.3

**Pull request merged**